mono/tests/threadpool-exceptions2.cs

```diff
@@ -2,38 +2,46 @@
 using System.Threading;
 
 class Test {
-	static object monitor;
-	static int return_value = 2;
+
+	static ManualResetEvent mre = new ManualResetEvent (false);
 
 	static int Main ()
 	{
-		monitor = new object ();
 		AppDomain.CurrentDomain.UnhandledException += OnUnhandledException;
 		WaitCallback wcb = new WaitCallback ((a) => {
 			throw new Exception ("From the threadpoool");
 		});
 		wcb.BeginInvoke (wcb, OnCBFinished, null);
-		lock (monitor) {
-			Monitor.Wait (monitor);
-		}
-		Thread.Sleep (1000);
-		return 1;
+
+		if (!mre.WaitOne (10000))
+			return 2;
+
+		GC.Collect ();
```

Comment:
What's the need for these 2 new function calls?

Reply:
They are not strictly necessary, could remove them but don't hurt either but it make the repro more reliable